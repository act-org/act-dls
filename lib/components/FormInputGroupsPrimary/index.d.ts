/**
 * @prettier
 */
import * as React from 'react';
interface FormInput {
    formInput: React.ReactElement<any>;
    key: string;
}
declare type FormInputGroup = FormInput[];
export interface Props {
    groups: FormInputGroup[];
}
declare const FormInputGroupsPrimary: React.FC<Props>;
export default FormInputGroupsPrimary;
//# sourceMappingURL=index.d.ts.map