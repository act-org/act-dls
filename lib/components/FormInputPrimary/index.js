"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _get = _interopRequireDefault(require("lodash/get"));

var _AlertOutline = _interopRequireDefault(require("../../icons/AlertOutline"));

var _colors = _interopRequireDefault(require("../../constants/colors"));

var _InputLabelPrimary = _interopRequireDefault(require("../InputLabelPrimary"));

var _InputPrimary = _interopRequireDefault(require("../InputPrimary"));

var _TooltipBase = _interopRequireDefault(require("../TooltipBase"));

var _styles = _interopRequireDefault(require("./styles"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var FormInputPrimary = function FormInputPrimary(_ref) {
  var disabled = _ref.disabled,
      errorMessage = _ref.errorMessage,
      inputProps = _ref.inputProps,
      labelProps = _ref.labelProps,
      required = _ref.required;
  var classes = (0, _styles["default"])();
  var children = /*#__PURE__*/React.createElement("div", {
    className: classes.inputContainer
  }, /*#__PURE__*/React.createElement(_InputPrimary["default"], _extends({
    disabled: disabled,
    error: !!errorMessage || (0, _get["default"])(inputProps, 'error'),
    fullWidth: true,
    required: required
  }, inputProps)), errorMessage && /*#__PURE__*/React.createElement(_TooltipBase["default"], {
    arrow: true,
    placement: "top",
    title: errorMessage
  }, /*#__PURE__*/React.createElement(_AlertOutline["default"], {
    classes: {
      root: classes.warning
    },
    htmlColor: _colors["default"].REQUIRED
  })));
  return /*#__PURE__*/React.createElement(React.Fragment, null, labelProps && /*#__PURE__*/React.createElement("div", {
    className: classes.labelContainer
  }, /*#__PURE__*/React.createElement(_InputLabelPrimary["default"], _extends({
    disabled: disabled,
    error: !!errorMessage || (0, _get["default"])(labelProps, 'error'),
    required: required
  }, labelProps))), children);
};

var _default = FormInputPrimary;
exports["default"] = _default;