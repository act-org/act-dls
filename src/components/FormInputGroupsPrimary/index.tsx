/**
 * @prettier
 */

import * as React from 'react';
import { Grid } from '@material-ui/core';

import useStyles from './styles';

interface FormInput {
  formInput: React.ReactElement<any>;
  key: string;
}

type FormInputGroup = FormInput[];

export interface Props {
  groups: FormInputGroup[];
}

const FormInputGroupsPrimary: React.FC<Props> = ({ groups }: Props) => {
  const classes = useStyles();

  return (
    <>
      {/* eslint-disable react/no-array-index-key */}
      {groups.map(
        (group, i): React.ReactElement<any> => (
          <Grid
            classes={{
              container: classes.gridContainerRoot,
            }}
            container
            key={i}
            spacing={4}
          >
            {group.map(
              (input): React.ReactElement<any> => (
                <Grid
                  classes={{
                    item: classes.gridItemRoot,
                  }}
                  item
                  key={input.key}
                >
                  {input.formInput}
                </Grid>
              ),
            )}
          </Grid>
        ),
      )}
      {/* eslint-enable react/no-array-index-key */}
    </>
  );
};

export default FormInputGroupsPrimary;
