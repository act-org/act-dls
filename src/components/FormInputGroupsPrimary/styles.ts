/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { makeStyles } from '@material-ui/core/styles';

import DIMS from '~/constants/dims';

export default makeStyles({
  gridContainerRoot: {
    marginBottom: DIMS.LAYOUT_PADDING / 2,
  },
  gridItemRoot: {
    flex: 1,
  },
});
