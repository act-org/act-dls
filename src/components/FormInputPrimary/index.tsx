/**
 * @prettier
 */

import * as React from 'react';
import { get } from 'lodash';
import { Tooltip } from '@material-ui/core';

import AlertOutline from '~/icons/AlertOutline';
import COLORS from '~/constants/colors';
import InputLabelPrimary, {
  Props as InputLabelPrimaryProps,
} from '~/components/InputLabelPrimary';
import InputPrimary, {
  Props as InputPrimaryProps,
} from '~/components/InputPrimary';

import useStyles from './styles';

export interface Props {
  disabled?: boolean;
  errorMessage?: string;
  inputProps?: InputPrimaryProps;
  labelProps?: InputLabelPrimaryProps;
  required?: boolean;
}

const FormInputPrimary: React.FC<Props> = ({
  disabled,
  errorMessage,
  inputProps,
  labelProps,
  required,
}: Props): React.ReactElement<Props> => {
  const classes = useStyles();

  const children = (
    <div className={classes.inputContainer}>
      <InputPrimary
        disabled={disabled}
        error={!!errorMessage || get(inputProps, 'error')}
        fullWidth
        required={required}
        {...inputProps}
      />

      {errorMessage && (
        <Tooltip arrow placement="top" title={errorMessage}>
          <AlertOutline
            classes={{
              root: classes.warning,
            }}
            htmlColor={COLORS.REQUIRED}
          />
        </Tooltip>
      )}
    </div>
  );

  return (
    <>
      {labelProps && (
        <div className={classes.labelContainer}>
          <InputLabelPrimary
            disabled={disabled}
            error={!!errorMessage || get(labelProps, 'error')}
            required={required}
            {...labelProps}
          />
        </div>
      )}

      {children}
    </>
  );
};

export default FormInputPrimary;
