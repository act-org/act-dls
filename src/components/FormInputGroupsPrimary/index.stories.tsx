/**
 * @prettier
 */

import * as React from 'react';

import FormInputPrimary from '~/components/FormInputPrimary';
import FormSelectPrimary from '~/components/FormSelectPrimary';

import FormInputGroupsPrimary from '.';

export const Primary = (): React.ReactElement<any> => {
  const [value1, setValue1] = React.useState('Value 1');
  const [value2, setValue2] = React.useState('Value 2');
  const [value3, setValue3] = React.useState('THREE');
  const [value4, setValue4] = React.useState('Value 4');
  const [value5, setValue5] = React.useState('Value 5');

  return (
    <FormInputGroupsPrimary
      groups={[
        [
          {
            formInput: (
              <FormInputPrimary
                inputProps={{
                  onChange: (e: any): void => {
                    setValue1(e.target.value);
                  },
                  placeholder: 'Input 1',
                  type: 'text',
                  value: value1,
                }}
                labelProps={{
                  children: 'Input 1',
                }}
                required
              />
            ),
            key: 'INPUT_1',
          },
          {
            formInput: (
              <FormInputPrimary
                inputProps={{
                  onChange: (e: any): void => {
                    setValue2(e.target.value);
                  },
                  placeholder: 'Input 2',
                  type: 'password',
                  value: value2,
                }}
                labelProps={{
                  children: 'Input 2',
                }}
                required
              />
            ),
            key: 'INPUT_2',
          },
        ],
        [
          {
            formInput: (
              <FormSelectPrimary
                labelProps={{
                  children: 'Input 3',
                }}
                options={[
                  {
                    label: 'One',
                    value: 'ONE',
                  },
                  {
                    label: 'Two',
                    value: 'TWO',
                  },
                  {
                    label: 'Three',
                    value: 'THREE',
                  },
                ]}
                required
                selectProps={{
                  onChange: (e: any): void => {
                    setValue3(e.target.value);
                  },
                  value: value3,
                }}
              />
            ),
            key: 'INPUT_3',
          },
        ],
        [
          {
            formInput: (
              <FormInputPrimary
                inputProps={{
                  onChange: (e: any): void => {
                    setValue4(e.target.value);
                  },
                  placeholder: 'Input 4',
                  type: 'number',
                  value: value4,
                }}
                labelProps={{
                  children: 'Input 4',
                }}
                required
              />
            ),
            key: 'INPUT_4',
          },
          {
            formInput: (
              <FormInputPrimary
                inputProps={{
                  onChange: (e: any): void => {
                    setValue5(e.target.value);
                  },
                  placeholder: 'Input 5',
                  type: 'text',
                  value: value5,
                }}
                labelProps={{
                  children: 'Input 5',
                }}
                required
              />
            ),
            key: 'INPUT_5',
          },
        ],
      ]}
    />
  );
};

export default {
  component: FormInputGroupsPrimary,
  parameters: {
    info: {
      text: 'This is the primary variant of the FormInputGroups component.',
    },
  },
  title: 'Organisms/FormInputGroups',
};
