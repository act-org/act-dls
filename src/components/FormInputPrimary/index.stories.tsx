/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { boolean, select, text } from '@storybook/addon-knobs';

import FormInputPrimary from '.';

export const Primary = (): React.ReactElement<any> => {
  const [value, setValue] = React.useState('Input');

  return (
    <FormInputPrimary
      inputProps={{
        autoFocus: boolean('Auto Focus?', false),
        disabled: boolean('Disabled?', false),
        onChange: (e: any): void => {
          action('onChange()');

          setValue(e.target.value);
        },
        placeholder: text('Placeholder', 'Placeholder'),
        type: select(
          'Type',
          {
            Password: 'password',
            Text: 'text',
          },
          'text',
        ),
        value,
      }}
      labelProps={{
        children: text('Label', 'Input Label'),
        disabled: boolean('Disabled?', false),
        error: boolean('Error?', false),
        required: boolean('Required?', true),
      }}
    />
  );
};

export default {
  component: FormInputPrimary,
  parameters: {
    info: {
      text: 'This is the primary variant of the FormInput component.',
    },
  },
  title: 'Molecules/FormInput',
};
