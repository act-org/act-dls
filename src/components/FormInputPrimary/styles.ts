/**
 * @prettier
 */

import { makeStyles } from '@material-ui/core/styles';

import DIMS from '~/constants/dims';

export default makeStyles({
  inputContainer: {
    position: 'relative',
  },
  labelContainer: {
    marginBottom: DIMS.LAYOUT_PADDING / 2,
  },
  warning: {
    marginLeft: 3,
    opacity: 0.5,
    position: 'absolute',
    right: -28,
    top: '35%',
  },
});
