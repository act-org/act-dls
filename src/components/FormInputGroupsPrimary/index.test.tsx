/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import FormInputPrimary from '~/components/FormInput';
import render from '~/helpers/test/render';

import FormInputGroupsPrimary from '.';

describe('FormInputGroupsPrimary', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <FormInputGroupsPrimary
        groups={[
          [
            {
              formInput: (
                <FormInputPrimary
                  inputProps={{
                    onChange: noop,
                    placeholder: 'Input 1',
                    type: 'text',
                    value: 'value1',
                  }}
                  labelProps={{
                    children: 'Input 1',
                  }}
                  required
                />
              ),
              key: 'INPUT_1',
            },
            {
              formInput: (
                <FormInputPrimary
                  inputProps={{
                    onChange: noop,
                    placeholder: 'Input 2',
                    type: 'password',
                    value: 'value2',
                  }}
                  labelProps={{
                    children: 'Input 2',
                  }}
                  required
                />
              ),
              key: 'INPUT_2',
            },
          ],
          [
            {
              formInput: (
                <FormInputPrimary
                  inputProps={{
                    onChange: noop,
                    placeholder: 'Input 3',
                    type: 'text',
                    value: 'value3',
                  }}
                  labelProps={{
                    children: 'Input 3',
                  }}
                />
              ),
              key: 'INPUT_3',
            },
          ],
        ]}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
