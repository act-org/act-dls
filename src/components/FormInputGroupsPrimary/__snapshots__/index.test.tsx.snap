// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputGroupsPrimary matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-1 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-2"
    >
      <div
        class="makeStyles-labelContainer-4"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-6 MuiInputLabel-animated"
        >
          <div
            class="label-split"
          >
            <span>
              Input 1
            </span>
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
      </div>
      <div
        class="makeStyles-inputContainer-3"
      >
        <div
          class="MuiInputBase-root MuiInput-root makeStyles-inputRoot-9 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-marginDense MuiInput-marginDense"
          required=""
        >
          <input
            class="MuiInputBase-input MuiInput-input makeStyles-inputInput-7 MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
            placeholder="Input 1"
            required=""
            type="text"
            value="value1"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-2"
    >
      <div
        class="makeStyles-labelContainer-4"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-6 MuiInputLabel-animated"
        >
          <div
            class="label-split"
          >
            <span>
              Input 2
            </span>
            <span
              class="required"
            >
              Required
            </span>
          </div>
        </label>
      </div>
      <div
        class="makeStyles-inputContainer-3"
      >
        <div
          class="MuiInputBase-root MuiInput-root makeStyles-inputRoot-9 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-marginDense MuiInput-marginDense"
          required=""
        >
          <input
            class="MuiInputBase-input MuiInput-input makeStyles-inputInput-7 makeStyles-inputInputPassword-8 MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
            placeholder="Input 2"
            required=""
            type="password"
            value="value2"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-gridContainerRoot-1 MuiGrid-spacing-xs-4"
  >
    <div
      class="MuiGrid-root MuiGrid-item makeStyles-gridItemRoot-2"
    >
      <div
        class="makeStyles-labelContainer-4"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-6 MuiInputLabel-animated"
        >
          <div
            class="label-split"
          >
            <span>
              Input 3
            </span>
          </div>
        </label>
      </div>
      <div
        class="makeStyles-inputContainer-3"
      >
        <div
          class="MuiInputBase-root MuiInput-root makeStyles-inputRoot-9 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-marginDense MuiInput-marginDense"
        >
          <input
            class="MuiInputBase-input MuiInput-input makeStyles-inputInput-7 MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
            placeholder="Input 3"
            type="text"
            value="value3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
